### PR TITLE
spotifyd: 0.2.11 -> 0.2.14

### PR DESCRIPTION
**Pull request by marsam**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @anderslundstedt

### CODE REVIEW

**Comment by anderslundstedt** [comment]
This fixes the build on Darwin, nice! Comments:

- Perhaps split this into two commits? One doing the version bump and one fixing the build on Darwin? At least indicate in commit message that this is not just a version bump.

- On MacOS I need to use the flag `--backend portaudio`, otherwise I get the error `Unknown backend: alsa`. I guess ALSA is not available for Darwin so perhaps it would make sense to create a wrapper on Darwin (that would simply run `spotifyd --backend portaudio $@`? We could merge this without that however.

File: pkgs/applications/audio/spotifyd/default.nix [hunk 1]
```diff
@@ -30,6 +30,8 @@ rustPlatform.buildRustPackage rec {
     ++ stdenv.lib.optional withPulseAudio libpulseaudio
     ++ stdenv.lib.optional withPortAudio portaudio;
 
+  doCheck = false;
```
Comment (anderslundstedt) [inline pkgs/applications/audio/spotifyd/default.nix]: Why this change?

**Comment by marsam** [comment]
@anderslundstedt thanks for reviewing
>Perhaps split this into two commits? 

sure, I've split it in two commits
> I guess ALSA is not available for Darwin so perhaps it would make sense to create a wrapper on Darwin 

I'm not sure about this, I would prefer this to handled upstream. I think https://github.com/Spotifyd/spotifyd/issues/297 is related.

File: pkgs/applications/audio/spotifyd/default.nix (hunk 1 above)
Comment (marsam) [reply pkgs/applications/audio/spotifyd/default.nix]: the checkPhase fails. I'm not sure why:
```
   ...
   Compiling spotifyd v0.2.14 (/private/tmp/nix-build-spotifyd-0.2.14.drv-0/source)
    Finished release [optimized] target(s) in 5m 20s
running tests
Running cargo test
error: the listed checksum of `/private/tmp/nix-build-spotifyd-0.2.14.drv-0/spotifyd-0.2.14-vendor/librespot-protocol/src/spirc.rs` has changed:
expected: 69d90e996fe6daf92580baa8bfe8a80b538c842b01505ec885e2e695c0460525
actual:   ed984f674231f831745048215c880f52929b004b65fc0bafe649d6516e701603

directory sources are not intended to be edited, if modifications are required then it is recommended that [replace] is used with a forked copy of the source
builder for '/nix/store/iy179xkb5wivnfh6mwfdqhnvsyd8z6nw-spotifyd-0.2.14.drv' failed with exit code 101
error: build of '/nix/store/iy179xkb5wivnfh6mwfdqhnvsyd8z6nw-spotifyd-0.2.14.drv' failed
```

**Comment by anderslundstedt** [comment]
> > I guess ALSA is not available for Darwin so perhaps it would make sense to create a wrapper on Darwin
> 
> I'm not sure about this, I would prefer this to handled upstream. I think [Spotifyd/spotifyd#297](https://github.com/Spotifyd/spotifyd/issues/297) is related.

Sure, makes sense.

File: pkgs/applications/audio/spotifyd/default.nix (hunk 1 above)
Comment (anderslundstedt) [reply pkgs/applications/audio/spotifyd/default.nix]: Neither am I. A quick grep shows that nixpkgs is full of `doCheck = false` so I think this solution is OK.

**Comment by anderslundstedt** [comment]
Apparently the command line options `cache_path` and `config` have been renamed to `cache-path` and `config-path`, respectively. Thus line 31 of the service file `nixos/modules/services/audio/spotifyd.nix` needs to be updated. Could you fix this?

**Comment by marsam** [comment]
nice catch! fixed

**Review by anderslundstedt — APPROVED** [review]
Everything looks good to me now!

**Review by anderslundstedt — APPROVED** [review]
